### ISSUE TITLE
Possible false negative on structural subtyping

### ISSUE THREAD

**Issue by MathiasSven** [microsoft/pyright#3325]
Pyright (CLI 1.1.235) reports no errors for
```py
from typing import Protocol


class Foo(Protocol):
    x: int
    y: str
    
class Bar(Protocol):
    x: bool
    y: str
    z: float

def f(a: Foo): ...

def g() -> Bar: ...

x = g()

f(x)
```
Mypy in this case fails to typecheck:
```py
error: Argument 1 to "f" has incompatible type "Bar"; expected "Foo"
note: Following member(s) of "Bar" have conflicts:
note:     x: expected "int", got "bool"
```
I am not sure what should be the correct behavior, but given that these fields are mutable, should it not fail to typecheck?

**Comment by erictraut** [comment]
Thanks for the bug report. Yes, I agree this is a type checking hole.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.236, which I just published. It will also be included in the next release of pylance.
